packages/cli/src/ui/hooks/usePrivacySettings.ts

```diff
@@ -0,0 +1,121 @@
+/**
+ * @license
+ * Copyright 2025 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { GaxiosError } from 'gaxios';
+import { useState, useEffect, useCallback } from 'react';
+import { Config, CodeAssistServer, UserTierId } from '@google/gemini-cli-core';
+
+export const usePrivacySettings = (config: Config) => {
+  const [dataCollectionOptIn, setDataCollectionOptIn] = useState<
+    boolean | undefined
+  >(undefined);
+  const [isFreeTier, setIsFreeTier] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+
+  const fetchDataCollectionOptIn = useCallback(async () => {
```

Comment:
inline this in the useEffect call on line 42 and add a cleanup return. otherwise if the config does change then this could get in a confusing state of a mixture of values set by the previous request and the new request.

Reply:
For discussion: this is somewhat tightly coupled to parts of core that would ideally be implementation details in the cli package. Is there a reasonable way we could make this less tightly coupled? One idea is we have good markdown rendering support in the cli package so the core package could have a generic method to return a privacy notice as markdown that the cli package renders rather than this code getting tightly coupled to implementation details of the core package.